<?xml version="1.0" encoding="UTF-8"?>
<!--
##
## Copyright (C) 2017-2025 Hanson Robotics
##
## This program is free software: you can redistribute it and/or modify
## it under the terms of the GNU General Public License as published by
## the Free Software Foundation, either version 3 of the License, or
## (at your option) any later version.
##
## This program is distributed in the hope that it will be useful,
## but WITHOUT ANY WARRANTY; without even the implied warranty of
## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
## GNU General Public License for more details.
##
## You should have received a copy of the GNU General Public License
## along with this program.  If not, see <https://www.gnu.org/licenses/>.
##
-->
<launch>
    <arg name="chatbot_node" default="true" />
    <arg name="chatbot_services_node" default="false" />
    <arg name="content_manager_node" default="false" />
    <arg name="rosbot_node" default="false" />
    <group ns="hr">
        <group ns="interaction">
            <node name="chatbot" pkg="ros_chatbot" type="run.py" respawn="true" if="$(arg chatbot_node)" output="screen">
                <param name="hybrid_response_topic" type="str" value="chatbot_responses" />
                <param name="response_chosen_topic" type="str" value="chatbot_response" />
                <param name="chat_topic" type="str" value="chat" />
                <param name="tts_ctrl_topic" type="str" value="/hr/control/speech/tts_control" />
                <param name="enable" type="bool" value="true" />
                <param name="hybrid_mode" type="bool" value="true" />
            </node>
            <node name="chatbot_services" pkg="ros_chatbot" type="services.py" respawn="true" if="$(arg chatbot_services_node)">
                <param name="image_topic" type="str" value="/hr/sensors/vision/chest_cam/chest_usb_cam/image_raw" />
            </node>
            <node name="content_manager" pkg="ros_chatbot" type="content_manager.py" respawn="true" if="$(arg content_manager_node)" />
            <node name="rosbot" pkg="ros_chatbot" type="rosbot.py" respawn="true" if="$(arg rosbot_node)" />
            <node name="priming" pkg="ros_chatbot" type="priming.py" respawn="true" if="$(arg chatbot_node)" />
        </group>
    </group>
</launch>
